### PR TITLE
docs: refactor style guide example 07-04

### PR DESCRIPTION
**Pull request by danielsogl**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Style guide example 07-04 uses the old Angular Http module.

## What is the new behavior?
Updated the example to use thew new HttpClient.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by danielsogl** [comment]
cc @brandonroberts

File: aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero.service.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // #docregion
 import { Injectable } from '@angular/core';
-
-import { of } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
Comment (brandonroberts) [inline aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero.service.ts]: This import should be from 'rxjs'

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero.service.ts (hunk 1 above)
Comment (danielsogl) [reply aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero.service.ts]: You are right!

**Comment by danielsogl** [comment]
@brandonroberts fixed

File: aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero-arena.service.avoid.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 import { HeroService } from './hero.service';
+
 // #docregion example
 /* avoid */
 
 export class HeroArena {
   constructor(
       @Inject(HeroService) private heroService: HeroService,
-      @Inject(Http) private http: Http) {}
+      @Inject(Http) private http: HttpClient) {}
```
Comment (gkalpak) [inline aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero-arena.service.avoid.ts]: `@Inject(Http)` should be `@Inject(HttpClient)`.

**Review by gkalpak — CHANGES_REQUESTED** [review]

File: aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero-arena.service.avoid.ts (hunk 2 above)
Comment (danielsogl) [reply aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero-arena.service.avoid.ts]: Makes sense.

**Comment by danielsogl** [comment]
@gkalpak fixed

**Review by gkalpak — APPROVED** [review]

**Comment by danielsogl** [comment]
@brandonroberts can you remove the pr-cleanup label?

**Review by brandonroberts — APPROVED** [review]

**Review by brandonroberts — CHANGES_REQUESTED** [review]
The `app.module.ts` in the `styleguide` folder needs to be updated to use `HttpClientModule` also.

**Review by brandonroberts — APPROVED** [review]

**Review by brandonroberts — CHANGES_REQUESTED** [review]
Remove all references to `@angular/http`

**Comment by danielsogl** [comment]
I try will do the changes later this day

**Comment by brandonroberts** [comment]
@danielsogl reopen this PR if you are still interested in landing this change.

**Pull request closed without merging**